gameApi/index.ts

```diff
@@ -260,19 +269,27 @@ export const proposeGame = async (
   rulesContractAddress: string,
   isPaid?: boolean,
 ): Promise<GameProposedEventObject> => {
-  console.log('proposeGame', { contract, rulesContractAddress });
+  console.log('GameAPI proposeGame:', contract, 'rulesContractAddress: ', rulesContractAddress);
   const value = ethers.BigNumber.from(10).pow(16);
-  let wallet = await getSessionWallet(await getSigner().getAddress());
-  const gasEstimated = await contract.estimateGas.proposeGame(rulesContractAddress, []);
+  let wallet = await getSessionWallet(await (await getSigner()).getAddress());
+  console.log('GameAPI acceptGame: seesionWallet = ', wallet);
 
-  const tx = await contract.proposeGame(rulesContractAddress, [wallet.address], {
+  const gasEstimated = await contract.estimateGas.proposeGame(rulesContractAddress, []);
+  console.log('GameAPI proposeGame: gasEstimated = ', gasEstimated, Number(gasEstimated));
+  const tx = contract.proposeGame(rulesContractAddress, [wallet.address], {
     gasLimit: gasEstimated.mul(2),
     value: isPaid ? value : null,
   });
-  console.log('tx', tx);
-  const rc = await tx.wait();
-  console.log('rc', rc);
-  const event = rc.events.find((event: { event: string }) => event.event === 'GameProposed');
+  
+  console.log('GameAPI proposeGame: tx = ', tx);
+  const txResult = await tx;
+  console.log('GameAPI proposeGame: txResult = ', txResult);
+  const rc = txResult.wait();
```

Comment:
https://pub.dev/documentation/flutter_web3/latest/ethers/TransactionReceipt-class.html

Is it TransactionReceipt? Please annotate it for readability.